cluster-autoscaler/simulator/clustersnapshot/predicate/predicate_snapshot.go

```diff
@@ -32,15 +32,17 @@ import (
 // SchedulerBasedPredicateChecker to check scheduler predicates.
 type PredicateSnapshot struct {
```

Comment:
Pleas modify/add unit tests in `cluster-autoscaler/simulator/clustersnapshot/predicate/predicate_snapshot_test.go` to cover the new logic.

Reply:
added tests for CSI as well.